spec/realtime/channel.test.js

```diff
@@ -336,6 +336,359 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
 		}
 	};
 
+	testOnAllTransports(_exports, 'attachWithChannelParamsBasicChannelsGet', function(realtimeOpts) { return function(test) {
+		test.expect(3);
+		var testName = 'attachWithChannelParamsBasicChannelsGet';
+		try {
+			var realtime = helper.AblyRealtime(realtimeOpts);
+			realtime.connection.on('connected', function() {
+				var params = {
+					modes: 'subscribe',
+					delta: 'vcdiff'
+				};
+				var channelOptions = {
+					params: params
+				};
+				var channel = realtime.channels.get(testName, channelOptions);
+				channel.attach(function(err) {
+					if(err)
+						test.ok(false, 'Attach failed with error: ' + displayError(err));
+					test.deepEqual(channel.channelOptions, channelOptions, 'Check requested channel options');
+					test.deepEqual(channel.params, params, 'Check result params');
+					test.deepEqual(channel.modes, ['subscribe'], 'Check result modes');
```

Comment:
Wouldn't it be better to have a test that checks the state from realtime in some way, as opposed to just checking the attributes on the channel are set.  Checking the channel attributes are set is not a reliable test to ensure the params/modes are sent to Realtime and are being received correctly.

Reply:
The attributes of the channel are set based on the `ATTACHED` response from the server, i.e. the assumption is Realtime received and processed these correctly provided it sent the correct response.

Reply:
>  i.e. the assumption is Realtime received and processed these correctly provided it sent the correct response.

Sure, but that's a flawed assumption, because someone else could refactor how Realtime sets these attributes, and then this test would pass mistakenly. That test is assuming how the internals of the library works, which is not how test should work IMHO.

Reply:
Fixed in https://github.com/ably/ably-js/pull/613/commits/804461e067147114f10062235374bdf9b95a0c08